src/platform/ESP32/OTAImageProcessorImpl.cpp

```diff
@@ -182,7 +295,25 @@ void OTAImageProcessorImpl::HandleFinalize(intptr_t context)
     }
 #endif // CONFIG_ENABLE_ENCRYPTED_OTA
 
+#if CONFIG_ENABLE_DELTA_OTA
+    esp_err_t err = esp_delta_ota_finalize(imageProcessor->mDeltaOTAUpdateHandle);
+    if (err != ESP_OK)
+    {
+        ESP_LOGE(TAG, "esp_delta_ota_finalize() failed (%s)!", esp_err_to_name(err));
+        esp_ota_abort(imageProcessor->mOTAUpdateHandle);
```

Comment:
here as well, we need to do `imageProcessor->ReleaseBlock();` and  post the event stating that download failed, ` PostOTAStateChangeEvent(DeviceLayer::kOtaDownloadFailed);`